### ISSUE TITLE
Search SDK: Make DocumentSearchResult and SearchResult properties public again

### ISSUE THREAD

**Issue by brjohnstmsft** [Azure/azure-sdk-for-net#1776]
Opening issue on behalf of jsgoupil:

> Little piece of feedback regarding this update. All setters for DocumentSearchResult and SearchResult are now internal. This is problematic to easily support an offline scenario or when Azure Search is down. Previously, I would hit my database where Azure Search usually hits the indexer, then I would create SearchResult and DocumentSearchResult similar to the one returned by the SDK.

> Now I need to create my own classes then transform the DocumentSearchResult and SearchResult from the SDK into new classes. Since none of those classes have interfaces, I end up with an ugly cast.

**Comment by brjohnstmsft** [comment]
I have some questions for JS about the scenario:

- Do you experience regular downtime with Azure Search to the point where redirecting your searches to the database is actually necessary?
- How do you actually execute searches against your database? Are you using SQL full-text search?
- How would you execute offline searches? Do you use Lucene?
- Would you need `FacetResult` properties to be publicly settable too?

The reason we made the properties internal was simply for encapsulation. It would be no big deal for us to change it back, but first I'd like to make sure there aren't bigger problems with what you're trying to do. It seems to me that trying to write an abstraction layer over Azure Search, a DB, and an offline search engine would justify the effort of defining your own result types anyway, since it is very likely you'd only want to expose a subset of the functionality (the least common denominator among the three data sources).

**Comment by jsgoupil** [comment]
Thank you for opening the issue.

1. I just started to use Azure Search, so I haven't experienced anything yet. However, I am a big fan of running my application in offline mode where I can work somewhere without internet and get a OK website while developing.
2. Basically, I provide a strip down version of Azure Search by running an ugly EntityFramework .Contains(..) on too many fields. Obviously, not awesome on performance but if something was going wrong at least my app is still running.
3. I am not using any Lucene.
4. At the moment, I am not using too many functionality, but I could see how it could be beneficial to let the user set those properties.

You are correct, at the moment, I have to define my own result types because of the internal properties. Not using facets, or other functionality from Azure leaves those properties null anyway. Most of the people will return the Azure Search classes directly.
The argument definitely can go either way. I return (before this internal update) DocumentSearchResult & SearchResult with all the scoring, highlights, etc. If I go offline, I was returning the same class but with scoring and highlight set to null.

Since the update, I made an exact copy of DocumentSearchResult & SearchResult that has public setters. I map all properties from the Azure instances to mine. In offline scenario I create my own DocumentSearchResult & SearchResult.

It's probably cleaner now, but to support this offline scenario (which I use probably 1% of the time) it's a big overkill.
